src/routes/upload.ts

```diff
@@ -1,36 +1,65 @@
 import { UPLOAD_PATH } from '../config';
 import { Router } from 'express';
-import { createWriteStream, existsSync, rmSync } from 'fs';
+import { createWriteStream, existsSync, mkdirSync, rmSync, statSync } from 'fs';
+const { execSync } = require('child_process');
 
 const router = Router();
+let fileSize: string | undefined;
 
 router.post('/', (req, res) => {
   try {
     req.pipe(req.busboy); // Pipe it trough busboy
 
+    req.busboy.on('field', (fieldname, file, data) => {
+      console.log(`Size of upload file is ${file}`);
+      fileSize = file;
+    });
     req.busboy.on('file', (fieldname, file, data) => {
       const filename = data && data.filename ? data.filename : 'upload.raucb';
       console.log(`Upload of '${filename}' started`);
       // Create a write stream of the new file
       try {
         try {
           if (existsSync(UPLOAD_PATH + filename)) {
-            rmSync(UPLOAD_PATH + filename)
+            const stat = statSync(UPLOAD_PATH + filename);
+
+            //Check if file is already existing and has the same size
+            if (fileSize && typeof fileSize === 'string' && stat && parseInt(fileSize) === stat.size) {
+              res.json({
+                output: 'done',
+              });
+            }
+            else {
+              rmSync(UPLOAD_PATH + filename);
+              const fstream = createWriteStream(UPLOAD_PATH + filename);
+              // Pipe it trough
+              file.pipe(fstream);
+
+              // On finish of the upload
+              fstream.on('close', () => {
+                console.log(`Upload of '${filename}' finished`);
+                res.json({
+                  output: 'done',
+                });
+              });
+            }
+          } else {
+            const fstream = createWriteStream(UPLOAD_PATH + filename);
+            // Pipe it trough
+            file.pipe(fstream);
+
+            // On finish of the upload
+            fstream.on('close', () => {
+              console.log(`Upload of '${filename}' finished`);
+              res.json({
+                output: 'done',
+              });
+            });
```

Comment:
Since the same write stream block is used twice here, I think we can make it cleaner in a way:
```
if (exists) {
  if (same file size) {
    res.json();
    return; // it will return here and will not get to uploading code part
  } else {
    rmSync()
  }
}
// and then uploading code:
const fstream = ..
```

Reply:
updated